src/notifications/hooks.ts

```diff
@@ -0,0 +1,59 @@
+import { useState, useMemo, useEffect, useCallback } from "react";
+import { useIntl } from "react-intl";
+import { useNotification, Notification } from "@reearth/state";
+
+export type NotificationStyleType = "error" | "warning" | "info" | "success";
+export type NotificationType = Notification;
+
+export default () => {
+  const intl = useIntl();
+  const [notification, setNotification] = useNotification();
+  const [visible, changeVisibility] = useState(false);
+  const notificationTimeout = 5000;
+
+  const notificationHeading = useMemo(() => {
+    if (!notification) {
+      return intl.formatMessage({ defaultMessage: "Notice" });
+    }
+    switch (notification.type) {
+      case "error":
+        return intl.formatMessage({ defaultMessage: "Error" });
+      case "warning":
+        return intl.formatMessage({ defaultMessage: "Warning" });
+      default:
+        return intl.formatMessage({ defaultMessage: "Notice" });
+    }
+  }, [intl, notification]);
```

Comment:
Messages are static value so you don't need to use useMemo:


```suggestion
  const errorMessage = intl.formatMessage({ defaultMessage: "Error" });
  const warningMessage = intl.formatMessage({ defaultMessage: "Warning" });
  const noticeMessage = intl.formatMessage({ defaultMessage: "Notice" });

  const notificationHeading = {
    error: errorMessage
    warning: warningMessage
  }[notification?.type] || noticeMessage;
```

Reply:
The above code doesn't work for a couple different reasons: notification.type is typed and won't allow JUST error and warning (it expects options for *info* and *success* as well). Also, since *notification* is possibly undefined, it doesn't like that.
I tried a couple other approaches but I can't find anything that works/works well like the useMemo.
Any other advice to keep this clean instead of me doing something like
```
const notificationHeading = notification?.type === "warning" ? intl.... : notification.type === "error" ? ..... etc etc
```

Reply:
> notification.type is typed and won't allow JUST error and warning (it expects options for info and success as well). Also, since notification is possibly undefined, it doesn't like that.

Currently there are only 3 messages for notification bar, so I think it is not a problem. Types except error and warning will become undefined, but then it will become "notice" message thanks to `|| noticeMessage`.

Also, an implementation using ternary operators is also acceptable.

Reply:
Sorry, I should have made clear I was getting type errors for the above code, even if I modified it a bit. So Typescript wasn't allowing me to use the above code. 
I'll go with a ternary operator. Thank you